components/scream/src/share/util/scream_time_stamp.cpp

```diff
@@ -65,8 +65,9 @@ TimeStamp::TimeStamp(const std::vector<int>& date,
   const auto sec  = time[2];
 
   // Check the days and seconds numbers are non-negative.
-  EKAT_REQUIRE_MSG (mm>0   && mm<=12, "Error! Month out of bounds.\n");
-  EKAT_REQUIRE_MSG (dd>0   && dd<=days_in_month(yy,mm), "Error! Day out of bounds.\n");
+  EKAT_REQUIRE_MSG (yy>=0   && yy<=9999, "Error! Year out of bounds.\n");
```

Comment:
Do we really need an upper bound on the year?  I'm sure we won't be running simulations for a millennium, so maybe this isn't an issue.  But it feels like an unnecessary limitation.  Who knows what the future holds in terms of simulations.

Reply:
Well, I wanted to put a check that caught possibly uninitialized inputs. No really compelling reasons though.